src/Microsoft.TemplateEngine.Abstractions/Installer/InstallResult.cs

```diff
@@ -10,32 +11,41 @@ namespace Microsoft.TemplateEngine.Abstractions.Installer
     /// </summary>
     public sealed class InstallResult : InstallerOperationResult
     {
-        private InstallResult(InstallRequest request, IManagedTemplatePackage templatePackage)
+        private InstallResult(InstallRequest request, IManagedTemplatePackage templatePackage, Dictionary<int, IList<string>>? vulnerabilities = null)
             : base(templatePackage)
         {
             InstallRequest = request;
+            Vulnerabilities = vulnerabilities;
         }
 
-        private InstallResult(InstallRequest request, InstallerErrorCode error, string errorMessage)
+        private InstallResult(InstallRequest request, InstallerErrorCode error, string errorMessage, Dictionary<int, IList<string>>? vulnerabilities = null)
              : base(error, errorMessage)
         {
             InstallRequest = request;
+            Vulnerabilities = vulnerabilities;
         }
 
         /// <summary>
         /// <see cref="InstallRequest"/> processed by <see cref="IInstaller.InstallAsync"/> operation.
         /// </summary>
         public InstallRequest InstallRequest { get; private set; }
 
+        /// <summary>
+        /// Vulnerabilities from installed package.
+        /// </summary>
+        /// It is a dictionary, as we don't want to add the NuGet Api package to this project.
+        public IReadOnlyDictionary<int, IList<string>>? Vulnerabilities { get; private set; }
```

Comment:
nullable dictionary may complicate the handling in caller and also imply that vulnerabilities were not checked. 
Consider different uses here.